### PR TITLE
Added install manual script

### PR DESCRIPTION
**Pull request by nisrulz**
I tested the script on my branch using below command:

```sh
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/nisrulz/JetBrainsMono/add/install-manual-script/install_manual.sh)
```

The URL is built like below:

`https://raw.githubusercontent.com/<username>/JetBrainsMono/<branch_name>/install_manual.sh`

The PR has the one setup for the JetBrains and master branch, in the readme.

PR Merge: Squash

### CODE REVIEW

File: install_manual.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+echo "Installing JetBrains Mono Fonts"
+echo ""
+# Get the download URL of the latest release from Github
+FONT_URL=$(curl -s https://api.github.com/repos/JetBrains/JetBrainsMono/releases/latest | grep browser_download_url | cut -d '"' -f 4)
```
Comment (dector) [inline install_manual.sh:6]: This will not work if a new asset will be added. 🤔

Comment (nisrulz) [reply install_manual.sh:6]: The script will be modified then, I believe 👀  As of now, there is only one asset 🤷🏻‍♂️

Comment (dector) [reply install_manual.sh:6]: Fair enough :) Unfortunately `jq` is not bundled in modern distributives.

Actually I wonder if setting up github action that will embed asset link might be an option here.

Comment (nisrulz) [reply install_manual.sh:6]: Where did I use `jq`🤔?

It uses the most basic and bundled tools on any linux distribution. I.e `curl`. `grep`, `cut` and `unzip`.

This is a a simple script.

I am not sure why would you want Github Actions setup here 🤔 Why overcomplicate things when they don't need to be?

Comment (dector) [reply install_manual.sh:6]: > Where did I use jqthinking?

Nowhere. My point was that `jq` is very handy for json processing in shell but it's not widespread and (unfortunately) can't be used in this case.

> I am not sure why would you want Github Actions setup here thinking Why overcomplicate things when they don't need to be?

Just making suggestions as current parsing method seems a bit hacky for me. :)
But I'm not a decision maker in this repo.

**Comment by bulenkov** [comment]
Hi @nisrulz. Having a link to https://raw.githubusercontent.com/nisrulz/JetBrainsMono/add/install-manual-script/install_manual.sh is a potential security vulnerability. How users of JetBrains Mono repo are protected from malicious modifications of the https://raw.githubusercontent.com/nisrulz/JetBrainsMono/add/install-manual-script/install_manual.sh file?

**Comment by nisrulz** [comment]
Lets break it down so it is easier to answer. There are many points when we start talking about malicious behaviour and how it propogates.

The link you mentioned in your comment is actually wrong. As mentioned in my PR description, I used it to test it. If you look at PR commit history, I do change it to one that would refer to Jetbrain owned repo i.e this repo.
Screenshot 2020-09-05 at 00 15 58

The specific commit is this one: https://github.com/JetBrains/JetBrainsMono/pull/288/commits/c4afddf38bdaed760b72d32eac36cd024fe19ac3

The url it now sets inside the `instal_manual.sh` file is this: `https://raw.githubusercontent.com/JetBrains/JetBrainsMono/master/install_manual.sh`

Note the username/repo name part: `https://raw.githubusercontent.com/`**JetBrains/JetBrainsMono**`/master/install_manual.sh`

What means is that once the PR is merged the `instal_manual.sh` being referenced comes from the `master` branch of **JetBrains/JetBrainsMono** repo, which is owned by Jetbrains.

Now since this is a manual install process, once merged, the script will be readable by all. Any changes done to it for malicious behaviour would need to me made as a PR by outside contributors (so be reviewed) ..or be committed by the Jetbrains engineers who have access to this repo.

Only in the case where Jetbrains engineers either commit directly or accept a malicious PR will it be an issue (which then goes into the territory of saying anything has a potential to be a malicious code inside of this repo).

If you do have an alternative solution that you think is better, please let me know. I use this one right now because I have automated setup for my laptop. Right now there is no direct way to install the font apart from manually downloading the font files.


Lets consider the above is the case, even in that case the script is clearly hosted on github and it can be read by anyone.

In the best case solution, we can get rid of the url and ask folks to clone the repo and execute the `shell` script from their own system.

**Comment by bulenkov** [comment]
@nisrulz Thank you for the clarification. I'll review the final version one more time and test it on a Linux VM. Any ideas where this sh will not work? What OSes have you tested already?

File: install_manual.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+# Extract the Zip filename
+FONT_ZIP_FILENAME=$(echo $FONT_URL | cut -d '/' -f 9)
+
+# Download the zip file from Github Releases using wget
+wget $FONT_URL 1>/dev/null 2>&1
+
+# Unzip and move to fonts directory
+unzip $FONT_ZIP_FILENAME -d ~/.local/share/fonts && fc-cache -fv
```
Comment (nisrulz) [inline install_manual.sh:15]: This is manual step line from readme.

File: install_manual.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+echo "Installing JetBrains Mono Fonts"
+echo ""
+# Get the download URL of the latest release from Github
+FONT_URL=$(curl -s https://api.github.com/repos/JetBrains/JetBrainsMono/releases/latest | grep browser_download_url | cut -d '"' -f 4)
+
+# Extract the Zip filename
+FONT_ZIP_FILENAME=$(echo $FONT_URL | cut -d '/' -f 9)
+
+# Download the zip file from Github Releases using wget
+wget $FONT_URL 1>/dev/null 2>&1
```
Comment (nisrulz) [inline install_manual.sh:12]: Download the font step using wget utility

File: README.md [hunk 4]
```diff
@@ -41,7 +41,7 @@ Select JetBrains Mono in the IDE settings: go to `Preferences/Settings` → `Edi
    - Linux. Open a terminal and run the following:
    
       ```bash
-      unzip <font_file.zip> -d ~/.local/share/fonts && fc-cache -fv
+      /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/JetBrains/JetBrainsMono/master/install_manual.sh)"
```
Comment (nisrulz) [inline README.md:44]: This is why the readme is modified to reflect that this script is to be run only for Linux.

**Comment by nisrulz** [comment]
I have tested this on 
- Ubuntu based distros i.e Ubuntu 20.04, 18.04 and Pop OS 20.04

This script is for linux based distros only. macOS has homebrew setup and windows has chocolatey.

Basically simplifying the step mentioned in the readme already:

Screenshot 2020-09-05 at 13 22 11

**Comment by bulenkov** [comment]
Tested on Ubuntu 20.04.

**Pull request merged**